File: ruby/_trip/trip_service.rb
require_relative '../trip/trip_dao'
require_relative '../user/user_session'
require_relative '../user/user'
require_relative '../еxceptions/user_not_logged_in_exception'
require_relative '../еxceptions/dependend_class_call_during_unit_test_exception'

class TripService

    def get_trip_by_user(user)
        trip_list = []
        logged_user = UserSession.get_instance.get_logged_user
        is_friend = false

        if(!logged_user.nil?)
            user.get_friends.each do |f|
                if(f.equal? logged_user)
                    is_friend = true
                    break
                end
            end

            if(is_friend)
                trip_list = TripDAO.find_trips_by_user(user)
            end

            return trip_list
        else
            raise UserNotLoggedInException.new
        end

    end

end

File: c++/c++_solution/cpp_solution/TripService.cpp
#include "pch.h"
#include "TripService.h"

#include <list>


std::list<Trip> TripService::GetTripsByUser(User& user)
{
	std::list<Trip> tripList;
	User* loggedUser = UserSession::GetInstance()->GetLoggedUser();
	bool isFriend = false;
	if (loggedUser)
	{
		std::list<User>::iterator i;
		for (i = user.GetFriends().begin(); i != user.GetFriends().end(); ++i)
		{
			if (*i == *loggedUser)
			{
				isFriend = true;
				break;
			}
		}
		if (isFriend)
		{
			tripList = TripDAO::FindTripsByUser(user);
		}
		return tripList;
	}
	else
	{
		throw UserNotLoggedInException("UserNotLoggedInException");
	}
}











//UserSession* UserSession::oneUserSession;
